config/routes.rb

```diff
@@ -28,7 +28,9 @@
       }
 
       resources :promotional_tweets, only: [:index, :create]
-      resources :subscriptions, only: [:create, :update]
+      resources :subscriptions, only: [:create, :update] do
+        post '/cancel' => 'subscriptions#cancel', on: :member
```

Comment:
It doesn't have to be but it's actually good for logging purposes to have it on the member. Also, I think you can simplify this like so:

```ruby
post :cancel, on: :member
```